product_default_image/models/res_company.py

```diff
@@ -0,0 +1,135 @@
+        ],
+        required=True,
+        default='none',
+        help='Determines the type of default images to use. '
+             'Changing this field will set all empty product images '
+             'to the target you specify. Products with a default image '
+             'from the previous target are changed to your new target '
+             'as well.\n\n'
+             'No Default Image: Deletes images.\n\n'
+             'Global: Set to Global Product Image.\n\n'
+             'Category: Set products to their category image.\n\n'
+             'Global and Category: Set products to their category '
+             'image if the category has an image. Otherwise set it to '
+             'the Global Product Image',
+    )
+    product_image = fields.Binary(
+        string='Global Product Image',
+        default=lambda s: s._default_product_image(),
+        help='Use as the global image for all product default images. '
+             'Limited to 1024x1024.',
+    )
+
+    @api.model
+    def _default_product_image(self):
+        image_path = get_module_resource(
+            'product_default_image', 'static/src/img', 'glob_prod_img.png'
+        )
+        with open(image_path, 'rb') as handler:
+            image_data = handler.read()
+        return tools.image_resize_image_big(
+            image_data.encode('base64')
+        )
+
+    @api.multi
+    def write(self, vals):
+        """ Changes product images if target has changed
+
+        Examples:
+
+            * In examples 1 and 2, product.template images
+              will not be updated with the new some_image
+              value.
+
+            * Examples 3 and 4 will ensure product.template images
+              are updated to the new some_image value.
+
+            .. code-block:: python
+
+            # 1
+            company.write({
+                'product_image': some_image,
+            })
+
+            # 2
+            company.product_image_target = 'global'
+            company.product_image = some_image
+
+            # 3
+            company.write({
+                'product_image': some_image,
+                'product_image_target': 'global',
+            })
+
+            # 4
+            company.product_image = some_image
+            company.product_image_target = 'global'
+
+        """
+        target = vals.get('product_image_target')
+
+        if 'product_image' not in vals and not target:
+            return super(ResCompany, self).write(vals)
+
+        if 'product_image' in vals:
+            vals['product_image'] = \
+                tools.image_resize_image_big(
+                    base64_source=vals['product_image'],
+                )
+
+        img_args = {
+            'from_types': [
+                NONE, GLOBAL, CATEGORY,
+            ],
+            'to_type': target,
+            'to_img_bg': None,
+            'in_cache': False,
+            'add_domain': None,
+        }
+
+        if target == GLOBAL_CATEGORY:
```

Comment:
This seems like some leftover code that needs to be dropped now that the logic at L126 is present.